mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala

```diff
@@ -113,6 +106,55 @@ class NaiveBayesModel private[mllib] (
     }
   }
 
+  def predictProbabilities(testData: RDD[Vector]): RDD[Map[Double, Double]] = {
+    val bcModel = testData.context.broadcast(this)
+    testData.mapPartitions { iter =>
+      val model = bcModel.value
+      iter.map(model.predictProbabilities)
+    }
+  }
+
+  def predictProbabilities(testData: Vector): Map[Double, Double] = {
+    modelType match {
+      case Multinomial =>
+        val prob = multinomialCalculation(testData)
+        posteriorProbabilities(prob)
+      case Bernoulli =>
+        val prob = bernoulliCalculation(testData)
+        posteriorProbabilities(prob)
+      case _ =>
```

Comment:
I didn't code that so didn't took any decision on that, but now that you highlighted it I think you're right. Removing the last case doesn't generate any compiler warning nor error so I just removed it.

I don't get any warning from my IDE but trying to run `sbt mllib/compile` gives me the following error:

```
[info] Compiling 15 Java sources to /media/SB-1TB/workarea/apache-spark/unsafe/target/scala-2.10/classes...
[error] /media/SB-1TB/workarea/apache-spark/unsafe/src/main/java/org/apache/spark/unsafe/memory/MemoryBlock.java:20: error: cannot find symbol
[error] import javax.annotation.Nullable;
[error]                        ^
[error]   symbol:   class Nullable
[error]   location: package javax.annotation
[error] /media/SB-1TB/workarea/apache-spark/unsafe/src/main/java/org/apache/spark/unsafe/memory/MemoryLocation.java:20: error: cannot find symbol
[error] import javax.annotation.Nullable;
[error]                        ^
[error]   symbol:   class Nullable
[error]   location: package javax.annotation
[error] /media/SB-1TB/workarea/apache-spark/unsafe/src/main/java/org/apache/spark/unsafe/memory/ExecutorMemoryManager.java:24: error: package javax.annotation.concurrent does not exist
[error] import javax.annotation.concurrent.GuardedBy;
[error]                                   ^
[error] /media/SB-1TB/workarea/apache-spark/unsafe/src/main/java/org/apache/spark/unsafe/memory/MemoryBlock.java:37: error: cannot find symbol
[error]   MemoryBlock(@Nullable Object obj, long offset, long length) {
[error]                ^
[error]   symbol:   class Nullable
[error]   location: class MemoryBlock
[error] /media/SB-1TB/workarea/apache-spark/unsafe/src/main/java/org/apache/spark/unsafe/memory/MemoryLocation.java:28: error: cannot find symbol
[error]   @Nullable
[error]    ^
[error]   symbol:   class Nullable
[error]   location: class MemoryLocation
[error] /media/SB-1TB/workarea/apache-spark/unsafe/src/main/java/org/apache/spark/unsafe/memory/MemoryLocation.java:33: error: cannot find symbol
[error]   public MemoryLocation(@Nullable Object obj, long offset) {
[error]                          ^
[error]   symbol:   class Nullable
[error]   location: class MemoryLocation
[error] /media/SB-1TB/workarea/apache-spark/unsafe/src/main/java/org/apache/spark/unsafe/memory/ExecutorMemoryManager.java:42: error: cannot find symbol
[error]   @GuardedBy("this")
[error]    ^
[error]   symbol:   class GuardedBy
[error]   location: class ExecutorMemoryManager
[error] 7 errors
[error] (unsafe/compile:compile) javac returned nonzero exit code
[error] Total time: 1 s, completed Jun 15, 2015 8:00:11 PM
```